apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -0,0 +1,200 @@
+
+@Injectable()
+export class UpsertWorkflowUseCase {
+  constructor(
+    private createWorkflowGenericUsecase: CreateWorkflowGeneric,
+    private updateWorkflowUsecase: UpdateWorkflow,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private notificationGroupRepository: NotificationGroupRepository,
+    private environmentRepository: EnvironmentRepository
+  ) {}
+  async execute(command: UpsertWorkflowCommand): Promise<WorkflowResponseDto> {
+    await this.validateEnvironment(command);
+    const existingWorkflow = await this.getWorkflowIfExist(command);
+    const workflow = await this.createOrUpdateWorkflow(existingWorkflow, command);
+
+    return WorkflowTemplateGetMapper.toResponseWorkflowDto(workflow);
+  }
+
+  private async createOrUpdateWorkflow(
+    existingWorkflow: NotificationTemplateEntity | null,
+    command: UpsertWorkflowCommand
+  ): Promise<NotificationTemplateEntity> {
+    if (existingWorkflow) {
+      return await this.updateWorkflowUsecase.execute(
+        UpdateWorkflowCommand.create(this.convertCreateToUpdateCommand(command, existingWorkflow))
+      );
+    }
+    const notificationGroupId = await this.getNotificationGroup(
+      command.workflowDto.notificationGroupId,
+      command.user.environmentId
+    );
+    if (!notificationGroupId) {
+      throw new BadRequestException('Notification group not found');
+    }
+    const isWorkflowActive = command.workflowDto?.active ?? true;
+
+    return await this.createWorkflowGenericUsecase.execute(
+      CreateWorkflowCommand.create(
+        this.buildCreateWorkflowGenericCommand(notificationGroupId, isWorkflowActive, command)
+      )
+    );
+  }
+
+  private persistWorkflow(workflowEntityToPersist: any) {
+    // this.notificationTemplateRepository.create({
+  }
+
+  private async validateEnvironment(command: UpsertWorkflowCommand) {
+    const environment = await this.environmentRepository.findOne({ _id: command.user.environmentId });
+
+    if (!environment) {
+      throw new BadRequestException('Environment not found');
+    }
+  }
+
+  private buildCreateWorkflowGenericCommand(
+    notificationGroupId: string,
+    isWorkflowActive: boolean,
+    command: UpsertWorkflowCommand
+  ): CreateWorkflowCommand {
+    const { user } = command;
+    const { workflowDto } = command;
+
+    return {
+      notificationGroupId,
+      draft: !isWorkflowActive,
+      environmentId: user.environmentId,
+      organizationId: user.organizationId,
+      userId: user._id,
+      name: command.workflowDto.workflowId || command.workflowDto.name,
+      __source: 'bridge',
+      type: WorkflowTypeEnum.BRIDGE,
+      origin: workflowDto.origin,
+      steps: this.mapSteps(workflowDto.steps),
+      controls: workflowDto.controls,
+      rawData: command as unknown as Record<string, unknown>,
+      payloadSchema: {},
+      active: isWorkflowActive,
+      description: workflowDto.description || 'Missing Description',
```

Comment:
I do not see any reason to store a missing description in the database, do you see any use cases for it?

```suggestion
      description: workflowDto.description || '',
```

Reply:
resolved during a sync convo

Reply:
@tatarco can you please document the reason here as a reply for posterity? It's important so that we can communicate async effectively.

It's not clear to me why this description is required right now - is it possible for this fallback to display in the client as an empty state?

Reply:
If I remember correctly, we agreed to provide a default mapping for our API responses for now. However, we not got to talk about setting a default string while storing it in the database.

Reply:
@rifont Legit, I'll try to get better, @djabarovgeorge I'm looking into it again

Reply:
removed it all together, mistake on my part,